packages/captp/src/captp.js

```diff
@@ -401,22 +489,131 @@ export function makeCapTP(ourId, rawSend, bootstrapObj = undefined, opts = {}) {
+  const rets = {
+    abort,
+    dispatch,
+    getBootstrap,
+    serialize,
+    unserialize,
+    makeTrapHandler,
+    Trap: /** @type {Trap | undefined} */ (undefined),
+  };
+
+  if (takeTrapReply) {
+    // Create the Trap proxy maker.
+    const makeTrapImpl = implMethod => (target, ...implArgs) => {
+      assert(
+        Promise.resolve(target) !== target,
+        X`Trap(${target}) target cannot be a promise`,
+      );
+
+      const slot = valToSlot.get(target);
+      assert(
+        slot && slot[1] === '-',
+        X`Trap(${target}) target was not imported`,
+      );
+      assert(
+        slot[0] === 't',
+        X`Trap(${target}) imported target was not created with makeTrapHandler`,
+      );
+      assert(
+        takeTrapReply,
+        X`Trap(${target}) failed; no opts.takeTrapReply supplied to makeCapTP`,
+      );
+
+      // Send a "trap" message.
+      lastQuestionID += 1;
+      const questionID = lastQuestionID;
+
+      // Encode the "method" parameter of the CTP_CALL.
+      let method;
+      switch (implMethod) {
+        case 'get': {
+          const [prop] = implArgs;
+          method = serialize(harden([prop]));
+          break;
+        }
+        case 'applyFunction': {
+          const [args] = implArgs;
+          method = serialize(harden([null, args]));
+          break;
+        }
+        case 'applyMethod': {
+          const [prop, args] = implArgs;
+          method = serialize(harden([prop, args]));
+          break;
+        }
+        default: {
+          assert.fail(X`Internal error; unrecognized implMethod ${implMethod}`);
+        }
+      }
+
+      assert(trapGuest, X`Trap(${target}) internal error; no trapGuest`);
+
+      // Set up the trap call with its identifying information.
+      const takeIt = takeTrapReply(implMethod, slot, implArgs);
+      return trapGuest.doTrap(
+        takeIt,
+        () =>
+          send({
+            type: 'CTP_CALL',
+            epoch,
+            trap: true, // This is the magic marker.
+            questionID,
+            target: slot,
+            method,
+          }),
+        trapBuf =>
+          send({
+            type: 'CTP_TRAP_NEXT_BUFFER',
+            epoch,
+            questionID,
+            trapBuf,
+          }),
+      );
+    };
+
+    /** @type {TrapImpl} */
+    const trapImpl = {
+      applyFunction: makeTrapImpl('applyFunction'),
+      applyMethod: makeTrapImpl('applyMethod'),
+      get: makeTrapImpl('get'),
+    };
+    harden(trapImpl);
+
+    rets.Trap = makeTrap(trapImpl);
+  }
+
+  return harden(rets);
 }
 
+/**
+ * @typedef {Object} LoopbackOpts
+ * @property {typeof defaultFar} Far
+ */
+
 /**
  * Create an async-isolated channel to an object.
  *
  * @param {string} ourId
- * @returns {{ makeFar<T>(x: T): ERef<T>, makeNear<T>(x: T): ERef<T>  }}
+ * @param {Partial<LoopbackOpts>} [opts]
+ * @returns {{ makeFar<T>(x: T): ERef<T>, makeNear<T>(x: T): ERef<T>,
+ * makeTrapHandler<T>(x: T): T, Trap: Trap }}
```

Comment:
In comments, we must do our own prettification
```suggestion
 * @returns {{
 *   makeFar<T>(x: T): ERef<T>,
 *   makeNear<T>(x: T): ERef<T>,
 *   makeTrapHandler<T>(x: T): T, 
 *   Trap: Trap
 * }}
```